server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/OdcDBStructureComparator.java

```diff
@@ -0,0 +1,146 @@
+import com.oceanbase.tools.dbbrowser.model.DBObjectType;
+import com.oceanbase.tools.dbbrowser.model.DBTable;
+import com.oceanbase.tools.dbbrowser.model.DBTable.DBTableOptions;
+import com.oceanbase.tools.dbbrowser.model.DBTableColumn;
+import com.oceanbase.tools.dbbrowser.model.DBTableConstraint;
+import com.oceanbase.tools.dbbrowser.model.DBTableIndex;
+import com.oceanbase.tools.dbbrowser.schema.DBSchemaAccessor;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author jingtian
+ * @date 2024/1/4
+ * @since ODC_release_4.2.4
+ */
+@Slf4j
+public class OdcDBStructureComparator implements DBStructureComparator {
+    private final List<DialectType> supportedDialectTypes =
+            Arrays.asList(DialectType.MYSQL, DialectType.OB_MYSQL, DialectType.OB_ORACLE);
+    private final List<DBObjectType> supportedDBObjectTypes = Arrays.asList(DBObjectType.TABLE);
+
+    @Override
+    public List<DBObjectComparisonResult> compare(@NonNull DBStructureComparisonConfig srcConfig,
+            @NonNull DBStructureComparisonConfig tgtConfig)
+            throws SQLException {
+        List<DBObjectComparisonResult> returnVal = new ArrayList<>();
+        checkConfig(srcConfig, tgtConfig);
+
+        DBSchemaAccessor srcAccessor = SchemaPluginUtil.getSchemaBrowserExtension(srcConfig.getDialectType())
+                .getDBSchemaAccessor(srcConfig.getDataSource().getConnection());
+        DBSchemaAccessor tgtAccessor = SchemaPluginUtil.getSchemaBrowserExtension(tgtConfig.getDialectType())
+                .getDBSchemaAccessor(tgtConfig.getDataSource().getConnection());
+        DBTableEditor tgtTableEditor = SchemaPluginUtil.getSchemaBrowserExtension(tgtConfig.getDialectType())
+                .getDBTableEditor(tgtConfig.getDataSource().getConnection());
+
+        Map<String, DBTable> srcTables = buildSchemaTables(srcAccessor, srcConfig.getSchemaName());
+        Map<String, DBTable> tgtTables = buildSchemaTables(tgtAccessor, tgtConfig.getSchemaName());
+        TableStructureComparator tableComparator = new TableStructureComparator(tgtTableEditor,
+                tgtConfig.getDialectType(),
+                srcConfig.getSchemaName(), tgtConfig.getSchemaName());
+
+        if (srcConfig.getBlackListMap().get(DBObjectType.TABLE) == null) {
+            /**
+             * Compare all the tables between source database and target database.
+             */
+            returnVal =
+                    tableComparator.compare(new ArrayList<>(srcTables.values()), new ArrayList<>(tgtTables.values()));
+        } else {
+            /**
+             * Compare specified tables between source database and target database.
+             */
+            for (String tableName : srcConfig.getBlackListMap().get(DBObjectType.TABLE)) {
+                if (srcTables.containsKey(tableName)) {
+                    returnVal.add(tableComparator.compare(srcTables.get(tableName), tgtTables.get(tableName)));
```

Comment:
Why do we compare tables in the blacklist?

Reply:
Because when initiating a structure comparison task, the comparison scope can be selected as "partial table"：
<img width="714" alt="image" src="https://github.com/oceanbase/odc/assets/140503120/bbb3f1ea-a4b0-4734-a27e-b53825fba0a9">

Reply:
what's the relation between 'blacklist' and 'partial table'? Does it mean that if a table is in the blacklist, we will not compare it?